runtime/runtime-params-estimator/src/gas_cost.rs

```diff
@@ -76,6 +76,21 @@ impl GasCost {
         result
     }
 
+    /// Like cmp::Ord::min but operates on heterogenous types (GasCost + Gas).
+    pub(crate) fn min_gas(mut self, gas: Gas) -> Self {
+        if self.to_gas() < gas {
+            let to_add = Ratio::from(gas - self.to_gas());
+            if let Some(qemu) = &mut self.qemu {
+                qemu.instructions += to_add / GAS_IN_INSTR;
+            } else if let Some(time_ns) = &mut self.time_ns {
+                *time_ns += to_add / GAS_IN_NS;
+            } else {
+                self.time_ns = Some(to_add / GAS_IN_NS);
+            }
```

Comment:
Why is this attempting to add a difference between the current and minimum gas, rather than just setting the minimum gas directly? This seems like it'd be prone to errors introduced by floor rounding, as a result of which `gas.min_gas(1_000_000).to_gas()` wouldn’t actually produce `1_000_000`.

I think at least a comment would be warranted.

Reply:
Yeah, adding is sub optimal for sure. For qemu values, I still fell like I have to do it because (in theory) it also measures the IO bytes and if those are non-zero I don't want to mess with them. I mean, _I could_ linearly increase all components but that's even more rounding error prone and I don't see any use case  where that would be helpful.

But for the time, setting it to the value directly just makes more sense.

Updated and added short comments.